examples/plot_function_identifier.py

```diff
@@ -3,13 +3,20 @@
 Identifying function names in a script
 ======================================
 
-Calls Sphinx-Gallery identify names function to figure out which
-functions are called in the script and to which module do they belong.
+This demonstrates how Sphinx-Gallery identifies names function to figure out
+which functions are called in the script and to which module do they belong.
+
+It uses both the code itself, as well as the docstrings (such as this one),
+as adding a reference to :func:`numpy.sin` and :func:`numpy.exp` will create
+proper backreferences even if they are not explicitly used. This is useful
+in particular when functions return classes -- if you add them to the docstring
```

Comment:
it seems from the regex below that this is true for any text block references yeah? Can we make this a bit clearer in the examples / documentation? They tend to refer to the docstring but this should also work within `#######` blocks right?